packages/dynamite/dynamite/lib/src/models/dynamite_config/config.dart

```diff
@@ -33,6 +33,66 @@ abstract class DynamiteConfig implements Built<DynamiteConfig, DynamiteConfigBui
 
   /// The specified line with the formatter should use.
   int? get pageWidth;
+
+  /// A collection of dynamite configs for a specific file.
+  ///
+  /// This parameter is only supported for the top level config.
+  // overrides is reserved in the build.yaml
+  BuiltMap<String, DynamiteConfig>? get overrides;
+
+  /// Whether the generated library should be marked with the `@experimental` annotation.
+  bool get experimental;
+
+  @BuiltValueHook(initializeBuilder: true)
+  static void _defaults(final DynamiteConfigBuilder b) {
+    b.experimental ??= false;
+  }
+
+  /// Gets the config for the given [uri].
+  ///
+  /// Returns this if no override is set.
+  DynamiteConfig configFor(final String uri) {
+    if (overrides == null) {
+      return this;
+    }
+
+    DynamiteConfig? override;
+    for (final entry in overrides!.entries) {
+      if (entry.key == uri) {
+        override = entry.value;
+        break;
+      }
+    }
+
```

Comment:
We might want to check here that the overrides do not contain overrides themselves

Reply:
it is a few lines below